tide_authenticated_content.module

```diff
@@ -237,3 +237,54 @@ function _tide_authenticated_content_user_role_validation($form, FormStateInterf
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function tide_authenticated_content_form_jsonapi_settings_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $route_config = \Drupal::config('tide_authenticated_content.route_settings');
+  $route_settings = $route_config->get("route_settings");
+  $form['tide_authenticated_content'] = [
+    '#type' => 'fieldset',
+    '#title' => t('Tide authenticated content'),
+    '#description' => t('Enable or disable the json routes.'),
+    '#collapsible' => TRUE,
+    '#collapsed' => FALSE,
+  ];
+  $route_list = [
+    'login' => 'Login',
+    'logout' => 'Logout',
+    'register' => 'Register',
+    'update' => 'Update',
+    'request_reset' => 'Request Reset',
+    'reset' => 'Reset Password',
+  ];
+  foreach ($route_list as $key => $route_name) {
+    $form['tide_authenticated_content'][$key] = [
+      '#type' => 'checkbox',
+      '#title' => $route_name,
+      '#default_value' => $route_settings[$key],
+    ];
+  }
+  $form['actions']['submit']['#submit'][] = 'tide_authenticated_content_form_jsonapi_settings_submit';
```

Comment:
`JsonapiExtrasSettingsForm` is a `ConfigFormBase`, please ensure that upon submitting the form, the config `jsonapi_extras.settings` does not have the submitted values of `tide_authenticated_content`.